metron-platform/metron-parsing/metron-parsers/src/test/java/org/apache/metron/parsers/leef/LEEFParserTest.java

```diff
@@ -0,0 +1,252 @@
+	 */
+	@Test
+	public void testLEEF_CEFlikeSample() {
+		List<JSONObject> parse = parse("LEEF:0|Incapsula|SIEMintegration|0|SQL Injection| fileId=3412364560000000008 sourceServiceName=test56111115.incaptest.co siteid=1333546 suid=300656 requestClientApplication=Mozilla/5.0 (Windows NT 6.1; WOW64; rv:38.0) Gecko/20100101 Firefox/38.0 popName=mia cs2=true cs2Label=Javascript Support cs3=true cs3Label=CO Support cs1=NA cs1Label=Cap Support cs4=936e64c2-bdd1-4719-9bd0-2d882a72f30d cs4Label=VID cs5=bab1712be85b00ab21d20bf0d7b5db82701f27f53fbac19a4252efc722ac9131fdc60c0da620282b02dfb8051e7a60f9 cs5Label=clappsig dproc=Browser cs6=Firefox cs6Label=clapp calCountryOrRegion=IL cicode=Rehovot cs7=31.8969 cs7Label=latitude cs8=34.8186 cs8Label=longitude Customer=siemtest start=1460303291788 url=test56111115.incaptest.co/ requestMethod=GET qstr=keywords\\=3%29%29%29%20AND%203434%3d%28%27%3amvc%3a%27%7c%7c%28SELECT%20CASE%203434%20WHEN%203434%20THEN%201%20ELSE%200%20END%20FROM%20RDB%24DATABASE%29%7c%7c%27%3aqvi%3a%27%29%20AND%20%28%28%283793%3d3793 cn1=200 proto=HTTP cat=REQ_PASSED deviceExternalId=2323800832649 dst=54.195.35.43 dstPort=80 in=406 xff=127.0.0.1 srcPort=443 src=127.0.0.1 protoVer=TLSv1.2 ECDHE-RSA-AES128-GCM-SHA256 fileType=12999,50999,50037,50044, filePermission=37,20,1,1, cs9=,High Risk SQL Expressions,,SQL SELECT Expression, cs9Label=Rule name");
+		JSONObject obj = parse.get(0);
+		Assert.assertNotNull(obj);
+		Assert.assertEquals("3412364560000000008", obj.get("fileId"));
+		Assert.assertEquals("Mozilla/5.0 (Windows NT 6.1; WOW64; rv:38.0) Gecko/20100101 Firefox/38.0", obj.get("requestClientApplication"));
+		Assert.assertTrue(obj.containsKey("longitude"));
+		Assert.assertFalse(obj.containsKey("cs8"));
+		Assert.assertFalse(obj.containsKey("cs8Label"));
+	}
+
+	@Test
+	public void testLEEFParserSample() throws Exception {
+		runTest("sample", Resources.readLines(Resources.getResource(getClass(), "sample.leef"), UTF_8),
+				Resources.toString(Resources.getResource(getClass(), "sample.schema"), UTF_8));
+	}
+
+	private void runTest(String name, List<String> lines, String schema) throws Exception {
+		runTest(name, lines, schema, "");
+	}
+
+	private void runTest(String name, List<String> lines, String schema, String targetJson) throws Exception {
+		for (String inputString : lines) {
+			JSONObject parsed = parse(inputString).get(0);
+			Assert.assertNotNull(parsed);
+			Assert.assertNotNull(parsed.get("timestamp"));
+			Assert.assertTrue((long) parsed.get("timestamp") > 0);
+
+			JSONParser parser = new JSONParser();
+
+			Map<?, ?> json = null;
+			try {
+				json = (Map<?, ?>) parser.parse(parsed.toJSONString());
+				Assert.assertEquals(true, validateJsonData(schema, json.toString()));
+			} catch (ParseException e) {
+				e.printStackTrace();
+			}
+
+			// test against an explicit json example
+			if (!targetJson.isEmpty()) {
+
```

Comment:
Are we testing anything here?

Reply:
Not really, this is a left over from a half complete idea in the current product's CEF Tests

Reply:
I will remove both